From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bradley Dice <bdice@bradleydice.com>
Date: Wed, 29 Jan 2025 12:00:00 -0800
Subject: [PATCH 2/2] Remove nostdlib flags to allow linking glibc

The conda-forge build environment provides glibc, so we can safely
link against it. This removes -nostdlib and -fno-builtin flags that
conflict with conda-forge's compiler flags (e.g. -fstack-protector).

---
 cext/CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/cext/CMakeLists.txt b/cext/CMakeLists.txt
index 7e032f5..59d3b7b 100644
--- a/cext/CMakeLists.txt
+++ b/cext/CMakeLists.txt
@@ -36,19 +36,19 @@ add_library(_cext_static STATIC
     tile_kernel.cpp
     )
 
-target_compile_options(_cext_static PUBLIC ${cext_compile_flags} ${nostdlib_flags})
+target_compile_options(_cext_static PUBLIC ${cext_compile_flags})
 target_include_directories(_cext_static PRIVATE ${cext_include_dirs})
-target_link_options(_cext_static PUBLIC ${cext_link_flags} ${nostdlib_flags})
+target_link_options(_cext_static PUBLIC ${cext_link_flags})
 
 
 # Build a Python extension
 
 add_library(_cext MODULE module.cpp)
 
-target_compile_options(_cext PUBLIC ${cext_compile_flags} ${nostdlib_flags})
+target_compile_options(_cext PUBLIC ${cext_compile_flags})
 target_include_directories(_cext PRIVATE ${cext_include_dirs})
 target_link_libraries(_cext PUBLIC _cext_static ${Python_LIBRARIES})
-target_link_options(_cext PUBLIC ${cext_link_flags} ${nostdlib_flags})
+target_link_options(_cext PUBLIC ${cext_link_flags})
 
 set_target_properties(_cext PROPERTIES SKIP_BUILD_RPATH TRUE)
 
@@ -65,9 +65,9 @@ endif()
 
 add_library(_cext_shared SHARED module.cpp)
 target_link_libraries(_cext_shared _cext_static ${Python_LIBRARIES} ${asan_library})
-target_compile_options(_cext_shared PUBLIC ${cext_compile_flags} ${nostdlib_flags})
+target_compile_options(_cext_shared PUBLIC ${cext_compile_flags})
 target_include_directories(_cext_shared PRIVATE ${cext_include_dirs})
-target_link_options(_cext_shared PUBLIC ${cext_link_flags} ${nostdlib_flags} -Wl,--no-undefined)
+target_link_options(_cext_shared PUBLIC ${cext_link_flags} -Wl,--no-undefined)
 
 # Tests
 
-- 
2.48.1

